bin/evm2wasm.js

```diff
@@ -2,16 +2,98 @@
 
 const evm2wasm = require('../index.js')
 const ethUtil = require('ethereumjs-util')
+const argv = require('minimist')(process.argv.slice(2))
+const fs = require('fs')
 
-const input = ethUtil.toBuffer(process.argv[2])
-
-evm2wasm.evm2wasm(input, {
-  stackTrace: process.argv[3] === 'trace',
-  tempName: 'temp',
-  inlineOps: true,
-  wabt: true
-}).then(function (output) {
-  console.log(output.toString('binary'))
-}).catch(function (err) {
-  console.error('Failed: ' + err)
-})
+//read EVM bytecode from a file
+function readEVM (file) {
+  return new Promise((resolve, reject) => {
+    if (file) {
+      fs.readFile(file, (err, data) => {
+        if (err) {
+          reject(err)
+        }
+
+        resolve(data) // strip newline
+      })
+    } else {
+      reject("no file")
+    }
+  })
+}
+
+//convert evm bytecode to WASM or WAST
+function convert (bytecode, wast) {
+  return new Promise((resolve, reject) => {
+    outputFile = argv.o ? argv.o : undefined
+
+    if(!bytecode) {
+      resolve(Buffer.from(''))
+    }
+
+    if (wast) {
+      let output = evm2wasm.evm2wast(bytecode, {
+        stackTrace: false,
+        tempName: 'temp',
+        inlineOps: true,
+        wabt: false 
+      })
+      resolve(output)
+    } else {
+      evm2wasm.evm2wasm(bytecode, {
+        stackTrace: false, 
+        tempName: 'temp',
+        inlineOps: true,
+        wabt: false
+      }).then(function (output) {
+        resolve(output)
+      }).catch(function (err) {
+        reject(err)
+      })
+    }
+  })
+}
+
+function storeOrPrintResult(output, outputFile) {
+  return new Promise((resolve, reject) => {
```

Comment:
Instead of this entire thing I think you could use `util.promisify(fs.writeFile)(...)`, see `wabt.js`